hnn_core/tests/test_network.py

```diff
@@ -956,3 +838,186 @@ def test_network_mesh():
     _ = jones_2009_model(mesh_shape=mesh_shape)
     _ = calcium_model(mesh_shape=mesh_shape)
     _ = law_2021_model(mesh_shape=mesh_shape)
+
+
+class TestPickConnection:
+    """Tests for the pick_connection function."""
+    @pytest.mark.parametrize("arg_name",
+                             ["src_gids", "target_gids", "loc", "receptor"]
+                             )
+    def test_1argument_none(self, base_network, arg_name):
+        """ Tests passing None as an argument value. """
+        net, _ = base_network
+        kwargs = {'net': net, f'{arg_name}': None}
+        indices = pick_connection(**kwargs)
+        assert not indices
+
+    @pytest.mark.parametrize("arg_name", ["src_gids", "target_gids"])
+    def test_1argument_range(self, base_network, arg_name):
+        """ Tests passing range as an argument value. """
+        net, _ = base_network
+        test_range = range(2)
+        kwargs = {'net': net, f'{arg_name}': test_range}
+        indices = pick_connection(**kwargs)
+
+        for conn_idx in indices:
+            assert set(test_range).issubset(
+                net.connectivity[conn_idx][arg_name]
+            )
+
+    @pytest.mark.parametrize("arg_name,value",
+                             [("src_gids", 'L2_pyramidal'),
+                              ("target_gids", 'L2_pyramidal'),
+                              ("loc", 'soma'),
+                              ("receptor", 'gabaa'),
+                              ])
+    def test_1argument_str(self, base_network, arg_name, value):
+        """ Tests passing string as an argument value. """
+        net, _ = base_network
+        kwargs = {'net': net, f'{arg_name}': value}
```

Comment:
directly pass `src_gids` etc instead of constructing f-strings? Think it would make the test cleaner